Sources/PackageModel/Manifest.swift

```diff
@@ -307,6 +313,20 @@ public final class Manifest {
 
         return packageDependency(referencedBy: packageName)
     }
+
+    /// Finds the package dependency referenced by the specified plugin usage.
+    /// - Returns: Returns `nil` if  the used plugin is from the same package or if the package the used plugin is from cannot be found.
+    public func packageDependency(
+        referencedBy pluginUsage: TargetDescription.PluginUsage) -> PackageDependency? {
+        switch pluginUsage {
+        case .plugin(_, let package):
```

Comment:
nit: can use `Some(package)` to eliminate the `if let package = package` dance, or map  on the optional if you prefer